cmd/cli/create/create.go

```diff
@@ -213,7 +213,12 @@ func create(cmd *cobra.Command, cmdArgs []string, OC *CreateOptions) error { //n
 	err = jobutils.VerifyJob(ctx, j)
 	if err != nil {
 		if _, ok := err.(*bacerrors.ImageNotFound); ok {
-			return fmt.Errorf("docker image '%s' not found in the registry, or needs authorization", j.Spec.Docker.Image)
+			// TODO(forrest): [fixme] the below error ought to be impossible.
+			dockerEngine, err := model.DockerEngineFromEngineSpec(j.Spec.EngineSpec)
```

Comment:
At present we get `Could not inspect image "ss" - could be due to repo/image not existing, or registry needing authorization` which is misleading because it's serving a dual purpose - we should be able to tell which it is.  This is probably the place we should be using ImageNotFound so that we can present a nicer and more accurate error message.